core/lib/types/src/kzg.rs

```diff
@@ -0,0 +1,315 @@
+    const SERIALIZED_SIZE: usize = 131_072 + 48 + 32 + 32 + 48 + 32 + 48;
+
+    /// Returns the bytes necessary for pubdata commitment part of batch commitments when blobs are used.
+    /// Return format: opening point (16 bytes) || claimed value (32 bytes) || commitment (48 bytes) || opening proof (48 bytes))
+    pub fn to_pubdata_commitment(&self) -> [u8; BYTES_PER_PUBDATA_COMMITMENT] {
+        let mut res = [0u8; BYTES_PER_PUBDATA_COMMITMENT];
+        // The crypto team/batch commitment expects the opening point to be 16 bytes
+        let mut truncated_opening_point = [0u8; 16];
+        truncated_opening_point.copy_from_slice(&self.opening_point.as_slice()[16..]);
+        res[0..16].copy_from_slice(&truncated_opening_point);
+        res[16..48].copy_from_slice(self.opening_value.as_slice());
+        res[48..96].copy_from_slice(self.kzg_commitment.as_slice());
+        res[96..144].copy_from_slice(self.opening_proof.as_slice());
+        res
+    }
+
+    /// Deserializes `Self::SERIALIZED_SIZE` bytes into `KzgInfo` struct
+    pub fn from_slice(data: &[u8]) -> Self {
+        assert_eq!(data.len(), Self::SERIALIZED_SIZE);
+
+        let mut ptr = 0;
+
+        let mut blob = [0u8; BYTES_PER_BLOB];
+        blob.copy_from_slice(&data[ptr..ptr + BYTES_PER_BLOB]);
+        ptr += BYTES_PER_BLOB;
+
+        let mut kzg_commitment = [0u8; 48];
+        kzg_commitment.copy_from_slice(&data[ptr..ptr + 48]);
+        ptr += 48;
+
+        let mut opening_point = [0u8; 32];
+        opening_point.copy_from_slice(&data[ptr..ptr + 32]);
+        ptr += 32;
+
+        let mut opening_value = [0u8; 32];
+        opening_value.copy_from_slice(&data[ptr..ptr + 32]);
+        ptr += 32;
+
+        let mut opening_proof = [0u8; 48];
+        opening_proof.copy_from_slice(&data[ptr..ptr + 48]);
+        ptr += 48;
+
+        let mut versioned_hash = [0u8; 32];
+        versioned_hash.copy_from_slice(&data[ptr..ptr + 32]);
+        ptr += 32;
+
+        let mut blob_proof = [0u8; 48];
+        blob_proof.copy_from_slice(&data[ptr..ptr + 48]);
+        ptr += 48;
+
+        assert_eq!(ptr, Self::SERIALIZED_SIZE);
+
+        Self {
+            blob: Blob::new(blob),
+            kzg_commitment: Bytes48::from_bytes(&kzg_commitment).unwrap(),
+            opening_point: Bytes32::new(opening_point),
+            opening_value: Bytes32::new(opening_value),
+            opening_proof: Bytes48::from_bytes(&opening_proof).unwrap(),
+            versioned_hash: Bytes32::new(versioned_hash),
+            blob_proof: Bytes48::from_bytes(&blob_proof).unwrap(),
+        }
+    }
+
+    /// Converts `KzgInfo` struct into a byte array
+    pub fn to_bytes(&self) -> [u8; Self::SERIALIZED_SIZE] {
+        let mut res = [0u8; Self::SERIALIZED_SIZE];
+
+        let mut ptr = 0;
+
+        res[ptr..ptr + BYTES_PER_BLOB].copy_from_slice(self.blob.as_slice());
```

Comment:
Probably using [nom](https://docs.rs/nom/latest/nom/) we can make it prettier

Reply:
we still need to allocate the arrays and copy into them. happy to make the update but im not sure if its prettier. for example: 

```
let (data, blob_bytes) = take::<usize, &[u8], ()>(BYTES_PER_BLOB)(data).unwrap();
let mut blob = [0u8; BYTES_PER_BLOB];
blob.copy_from_slice(blob_bytes);
```